### DISCUSSION TITLE
How to deal with data from useState or other hook

### DISCUSSION THREAD

**Discussion by lovebuizel** [statelyai/xstate#2663] [Q&A] [answered]
Goal:
The action can call outer function to console count value updated.

Problem:
When the count value updated, action still console initial value instead of latest value.
I know I can store count value in the machine to update and use, but I just want the value from useState or other hooks.
Thanks.

Code:
```
import { useState } from "react";
import { useMachine } from "@xstate/react";
import { createMachine } from "xstate";
import "./styles.css";

export default function App() {
  const [count, setCount] = useState(10);
  const handleClick = () => {
    console.log(count);
  };
  const [current, send] = useMachine(
    createMachine(
      {
        id: "test",
        initial: "ready",
        states: {
          ready: {
            on: {
              CLICK: {
                actions: ["click"]
              }
            }
          }
        }
      },
      {
        actions: {
          click: handleClick
        }
      }
    )
  );

  return (
    <div className="App">
      <div>{count}</div>
      <button onClick={() => setCount(count + 1)}>addCount</button>
      <button onClick={() => send("CLICK")}>printCount</button>
    </div>
  );
}
```
https://codesandbox.io/s/infallible-jones-ygjml?file=/src/App.js

**Comment by Andarist** [comment, marked as answer]
In such a case you should configure your actions in the second argument of the `useMachine` and not `createMachine`, see corrected example: https://codesandbox.io/s/little-sun-xgcyg?file=/src/App.js

**Reply by lovebuizel** [reply]
Wow, thanks for replying immediately. I got it.

**Comment by mattpocock** [comment]
@lovebuizel See also the documentation on how to work with other hooks:

https://xstate.js.org/docs/recipes/react.html#other-hooks

**Reply by lovebuizel** [reply]
Thanks.
